src/Tools/Test Framework/Test Runner/src/DataDrivenTest/DataInputs/TestInputsManagement.Codeunit.al

```diff
@@ -0,0 +1,284 @@
+            CurrentLineNo := ExistingTestMethodLine."Line No." + this.GetIncrement()
+        else
+            CurrentLineNo := this.GetIncrement();
+
+        ALTestSuite.Get(TestMethodLine."Test Suite");
+
+        repeat
+            this.TransferToTemporaryTestLine(TempTestMethodLine, ExistingTestMethodLine, CurrentLineNo, TestInput);
+            this.InsertTestMethodLines(TempTestMethodLine, ALTestSuite);
+            this.UpdateCodeunitTestInputProperties(TempTestMethodLine, TestInput);
+            TempTestMethodLine.DeleteAll();
+        until TestInput.Next() = 0;
+    end;
+
+    local procedure TransferToTemporaryTestLine(var TempTestMethodLine: Record "Test Method Line" temporary; var TestMethodLine: Record "Test Method Line"; var CurrentLineNo: Integer; var TestInput: Record "Test Input")
+    begin
+        TempTestMethodLine.TransferFields(TestMethodLine);
+        TempTestMethodLine."Line No." := CurrentLineNo;
+        CurrentLineNo += this.GetIncrement();
+        TempTestMethodLine."Data Input Group Code" := TestInput."Test Input Group Code";
+        TempTestMethodLine."Data Input" := TestInput.Code;
+        TempTestMethodLine.Insert();
+    end;
+
+    local procedure UpdateCodeunitTestInputProperties(var TempTestMethodLine: Record "Test Method Line" temporary; var DataInput: Record "Test Input")
+    var
+        CodeunitTestMethodLine: Record "Test Method Line";
+        LastTestMethodLine: Record "Test Method Line";
+        TestSuiteManagement: Codeunit "Test Suite Mgt.";
+    begin
+        LastTestMethodLine.SetRange("Test Suite", TempTestMethodLine."Test Suite");
+        LastTestMethodLine.FindLast();
+        CodeunitTestMethodLine.SetRange("Test Suite", TempTestMethodLine."Test Suite");
+        CodeunitTestMethodLine.SetRange("Line Type", CodeunitTestMethodLine."Line Type"::Codeunit);
+
+        CodeunitTestMethodLine.SetRange("Data Input Group Code", '');
+        CodeunitTestMethodLine.SetFilter("Line No.", TestSuiteManagement.GetLineNoFilterForTestCodeunit(LastTestMethodLine));
+        CodeunitTestMethodLine.ModifyAll("Data Input Group Code", DataInput."Test Input Group Code");
+        CodeunitTestMethodLine.SetRange("Data Input Group Code");
+
+        CodeunitTestMethodLine.SetRange("Data Input", '');
+        CodeunitTestMethodLine.ModifyAll("Data Input", DataInput.Code);
+    end;
+
+    procedure UploadAndImportDataInputsFromJson()
+    var
+        TestInputGroup: Record "Test Input Group";
+    begin
+        this.UploadAndImportDataInputsFromJson(TestInputGroup);
+    end;
+
+    internal procedure UploadAndImportDataInputsFromJson(FileName: Text; TestInputInStream: InStream)
+    var
+        TestInputGroup: Record "Test Input Group";
+        InputText: Text;
+    begin
+        if not TestInputGroup.Find() then
+            this.CreateTestInputGroup(TestInputGroup, FileName);
+
+        if FileName.EndsWith(this.JsonFileExtensionTxt) then begin
```

Comment:
We should check the file type/MIME using FileManagment.

Reply:
Wont Fix - we cannot use any dependencies in the test tool.